server/src/permissions/rule.js

```diff
@@ -40,6 +40,16 @@ class Ruleset {
     this._rules = rules;
   }
 
+  validation_required() {
+    for (const rule of this._rules) {
+      if (!rule._validator) {
+        return false;
```

Comment:
Is this right?  It seems like we need to do validation if *any* rule has a validator.